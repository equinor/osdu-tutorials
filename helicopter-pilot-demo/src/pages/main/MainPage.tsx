import React from "react";
import {authProvider} from "../../authProvider";

import "./style.css";


export function MainPage() {
    const {userName} = authProvider.getAccount();

    const accountUi = () => {
        if (userName === null) {
            return (
                <a className='auth-button' href="/">
                    Login
                </a>
            );
        } else {
            return (
                <>
                    <span>Welcome {userName}</span>
                    <a className='auth-button' href={`/logout?frontend_state=${encodeURIComponent(window.location.href)}`}>
                        Logout
                    </a>
                </>
            );
        }
    };

    return (
        <div className="main">
            <div className="auth-buttons">
                {accountUi()}
            </div>
        </div>
    );
}