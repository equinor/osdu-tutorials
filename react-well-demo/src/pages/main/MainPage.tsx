import React, { useState } from "react";
import "./styles.css";
import WellLog from "../../components/welllog/welllog";
import WellCanvas from "../../components/well/WellCanvas";
import Search from "../../components/well/Search";
import Account from "../../components/account";
import { Box } from "@mui/material";

/**
 * Contains login-logout functionality, search wells form,
 * found wells list and area for drawing well trajectories
 */
export function MainPage() {
  const [searchName, setSearchName] = useState<string>("");

  return (
    <div className="main">
      <div className="main__page">
        <Box className="navbar">
          <Search setSearchNameCallback={setSearchName} />
          <Account />
        </Box>
        <WellCanvas searchName={searchName} />
        <WellLog />
      </div>
    </div>
  );
}
