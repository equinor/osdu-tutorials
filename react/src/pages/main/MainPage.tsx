import React from 'react';
import './styles.css';
import { Search } from 'components/well-search';
import { authProvider } from '../../authProvider';

/**
 * Contains login-logout functionality, search wells form,
 * found wells list and area for drawing well trajectories
 */
export function MainPage() {
  const { name } = authProvider.getAccount();
  console.log(name);

  const accountUi = () => {
    if (name === null) {
      return (
        <a
          className='auth-button'
          >
          Login
        </a>
      );
    } else {
      return (
        <>
          <span>Welcome {name}</span>
          <a className='auth-button' href={`/logout?frontend_state=${encodeURIComponent(window.location.href)}`}>
            Logout
          </a>
        </>
      );
    }
  };

  return (
    <div className='main'>
      <div className='main__page'>
        {/* wells search with results */}
        <Search />

        {/* map canvas */}

        {/* login-logout functionality. plain and simple */}
        <div className='main__auth-buttons'>
          {accountUi()}
        </div>
      </div>
    </div>
  );
}
