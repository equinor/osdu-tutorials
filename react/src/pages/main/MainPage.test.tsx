import React from 'react';
import { shallow } from 'enzyme';
import { MainPage, MainPageProps } from './MainPage';

describe('MainPage', () => {
  it('should render self and subcomponents', () => {
    const props: MainPageProps = {
      isTrajectoryLoading: false,
      trajectoriesToDraw: [],
    };
    const wrapper = shallow(<MainPage {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
