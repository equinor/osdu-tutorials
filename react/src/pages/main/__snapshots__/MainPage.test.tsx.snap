// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage should render self and subcomponents 1`] = `
<div
  className="main"
>
  <div
    className="main__page"
  >
    <ConnectedSearch />
    <div
      className="main__chart-area"
    >
      <Hint
        subTitle="Find well and click visualize to appropriate data set"
        title="No Trajectory to display"
      />
    </div>
    <div
      className="main__auth-buttons"
    >
      <a
        className="auth-button"
        href="/login?frontend_state=http%3A%2F%2Flocalhost%2F"
      >
        Login
      </a>
      <a
        className="auth-button"
        href="/logout?frontend_state=http%3A%2F%2Flocalhost%2F"
      >
        Logout
      </a>
    </div>
  </div>
</div>
`;
